src/EnergyPlus/UnitarySystem.cc

```diff
@@ -1886,6 +1886,44 @@ namespace UnitarySystems {
 
             SizingMethod = DataHVACGlobals::SystemAirflowSizing;
 
+            MSHPIndex = this->m_DesignSpecMSHPIndex;
+            // set no load air flow ratio local var
+            Real64 NoLoadCoolingAirFlowRateRatio = 1.0;
+            Real64 NoLoadHeatingAirFlowRateRatio = 1.0;
+            if (MSHPIndex > -1) {
+                if (designSpecMSHP[MSHPIndex].coolingVolFlowRatio[0] == DataSizing::AutoSize) {
+                    NoLoadCoolingAirFlowRateRatio = min(this->m_NoLoadAirFlowRateRatio, 1.0 / double(designSpecMSHP[MSHPIndex].numOfSpeedCooling));
+                } else {
+                    NoLoadCoolingAirFlowRateRatio = min(this->m_NoLoadAirFlowRateRatio, designSpecMSHP[MSHPIndex].coolingVolFlowRatio[0]);
+                }
+                if (designSpecMSHP[MSHPIndex].heatingVolFlowRatio[0] == DataSizing::AutoSize) {
+                    NoLoadHeatingAirFlowRateRatio = min(this->m_NoLoadAirFlowRateRatio, 1.0 / double(designSpecMSHP[MSHPIndex].numOfSpeedHeating));
+                } else {
+                    NoLoadHeatingAirFlowRateRatio = min(this->m_NoLoadAirFlowRateRatio, designSpecMSHP[MSHPIndex].heatingVolFlowRatio[0]);
+                }
+                this->m_NoLoadAirFlowRateRatio = min(NoLoadCoolingAirFlowRateRatio, NoLoadHeatingAirFlowRateRatio);
+            } else {
+                if (this->m_CoolingCoilType_Num == DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed ||
+                    this->m_HeatingCoilType_Num == DataHVACGlobals::Coil_HeatingAirToAirVariableSpeed) {
```

Comment:
OK, so now what happens with multispeed and multistage coils and the new cooling coil model (Coil:Cooling:DX)?

Reply:
In all multi/variable speed coil test cases that I run the no-load mass flow rate is minimum flow rate, not the full flow rate. This change allows to determine the minimum flow rate either based on the minimum flow fraction autosized (or hard value user-input) when it is uses multi-speed coils, or based on the flow fraction (ratio of lowest speed flow to maximum speed flow) for variable speed DX coils. This change works fine with the new DX cooling coil as well. One of the unit tests uses the new DX cooling coil. Three of the defect files use the new dx cooling coil model.

Reply:
Then this seems to cover all cases.

Reply:
Below is expected diffs from one of the regression tests file (UnitarySystem_MultiSpeedDX.idf) which has a single speed gas heating coil and two speed new DX cooling coil. The no load flow rate is autosized to half of the full flow rate as shown below.

```
@@ -207,7 +207,7 @@
  Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 1 FURNACE DX COOL UNITARY SYSTEM, Design Size Supply Air Flow Rate [m3/s], 0.24830
  Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 1 FURNACE DX COOL UNITARY SYSTEM, Design Size Heating Supply Air Flow Rate [m3/s], 0.24830
  Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 1 FURNACE DX COOL UNITARY SYSTEM, Design Size Cooling Supply Air Flow Rate [m3/s], 0.24830
- Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 1 FURNACE DX COOL UNITARY SYSTEM, Design Size No Load Supply Air Flow Rate [m3/s], 0.24830
+ Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 1 FURNACE DX COOL UNITARY SYSTEM, Design Size No Load Supply Air Flow Rate [m3/s], 0.12415
  Component Sizing Information, Coil:Cooling:DX:CurveFit:OperatingMode, SYS 1 FURNACE DX COOL COOLING COIL OPERATING MODE, Design Size Rated Evaporator Air Flow Rate, 0.24830
  Component Sizing Information, Coil:Cooling:DX:CurveFit:OperatingMode, SYS 1 FURNACE DX COOL COOLING COIL OPERATING MODE, Design Size Rated Gross Total Cooling Capacity, 6165.88337
  Component Sizing Information, Coil:Cooling:DX:CurveFit:OperatingMode, SYS 1 FURNACE DX COOL COOLING COIL OPERATING MODE, User-Specified Rated Condenser Air Flow Rate, 0.00000

```

Reply:
For three speed new dx cooling coil and a single speed electric heating coil, the no-load flow rate sized to 1/3 of the full flow rate as expected and is shown below:
```
@@ -230,7 +230,7 @@
  Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size Supply Air Flow Rate [m3/s], 0.19125
  Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size Heating Supply Air Flow Rate [m3/s], 0.19125
  Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size Cooling Supply Air Flow Rate [m3/s], 0.19125
- Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size No Load Supply Air Flow Rate [m3/s], 0.19125
+ Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size No Load Supply Air Flow Rate [m3/s], 6.37490E-002
  Component Sizing Information, Coil:Cooling:DX:CurveFit:OperatingMode, SYS 2 FURNACE DX COOL COOLING COIL OPERATING MODE, Design Size Rated Evaporator Air Flow Rate, 0.19125
  Component Sizing Information, Coil:Cooling:DX:CurveFit:OperatingMode, SYS 2 FURNACE DX COOL COOLING COIL OPERATING MODE, Design Size Rated Gross Total Cooling Capacity, 3958.37344
  Component Sizing Information, Coil:Cooling:DX:CurveFit:OperatingMode, SYS 2 FURNACE DX COOL COOLING COIL OPERATING MODE, User-Specified Rated Condenser Air Flow Rate, 0.00000
```

Reply:
I think so based on the tests I have run. I have tested UnitarySystem with  UnitarySystemPerformance:Multispeed objects specified for various number of the new DX cooling coil speeds and the old multispeedDXcoils and single speed coils. Also test the variable speed DX coils without UnitarySystemPerformance:Multispeed objects. All behaved as expected.